prover/prover.go

```diff
@@ -257,17 +257,98 @@ func InitFromConfig(ctx context.Context, p *Prover, cfg *Config) (err error) {
 	if p.IsGuardianProver() {
 		proverServerOpts.ProverPrivateKey = p.cfg.GuardianProverPrivateKey
 	}
+
 	if p.srv, err = server.New(proverServerOpts); err != nil {
 		return err
 	}
 
 	return nil
 }
 
+// setApprovalAmount will set the allowance on the TaikoToken contract for the
+// configured proverAddress as owner and the TaikoL1 contract as spender,
+// if flag is provided for allowance.
+func (p *Prover) setApprovalAmount() error {
+	if p.cfg.Allowance == nil || p.cfg.Allowance.Cmp(common.Big0) != 1 {
+		log.Info("skipping setting approval, allowance not set")
+		return nil
+	}
+
+	allowance, err := p.rpc.TaikoToken.Allowance(
+		&bind.CallOpts{},
+		p.proverAddress,
+		p.cfg.TaikoL1Address,
+	)
+	if err != nil {
+		return err
+	}
+
+	log.Info("existing allowance for taikoL1 contract", "allowance", allowance.String())
+
+	if allowance.Cmp(p.cfg.Allowance) == 1 {
+		log.Info("skipping setting allowance, allowance already greater or equal",
+			"allowance", allowance.String(),
+			"approvalAmount", p.cfg.Allowance.String(),
+		)
+		return nil
+	}
+
+	opts, err := bind.NewKeyedTransactorWithChainID(
+		p.cfg.L1ProverPrivKey,
+		p.rpc.L1ChainID,
+	)
+	if err != nil {
+		return err
+	}
+
+	amt := new(big.Int).Exp(big.NewInt(1_000_000_000), big.NewInt(18), nil)
+
+	log.Info("prover approving taikoL1 for taiko token", "amt", amt.String())
+
+	tx, err := p.rpc.TaikoToken.Approve(
+		opts,
+		p.cfg.TaikoL1Address,
+		amt,
+	)
+	if err != nil {
+		return err
+	}
+
+	receipt, err := rpc.WaitReceipt(context.Background(), p.rpc.L1, tx)
+	if err != nil {
+		return err
+	}
+
+	if receipt.Status != 1 {
```

Comment:
Looks like `status` has already been checked in `rpc.WaitReceipt(context.Background(), p.rpc.L1, tx)`

Reply:
removed it!